exporter/opentelemetry-exporter-prometheus-remote-write/README.rst

```diff
@@ -1,27 +1,248 @@
+values in the ``headers`` dictionary
+
+1. Basic Authentication Basic authentication sets a HTTP Authorization
+   header containing a base64 encoded username/password pair. See `RFC
+   7617`_ for more information. This
+
+.. code:: python
+
+   exporter = PrometheusRemoteWriteMetricsExporter(
+       basic_auth={"username": "base64user",  "password": "base64pass"}
+   )
+
+2. Bearer Token Authentication This custom configuration can be achieved
+   by passing in a custom ``header`` to the constructor. See `RFC 6750`_
+   for more information.
+
+.. code:: python
+
+   header = {
+       "Authorization": "Bearer mytoken123"
+   }
+
+TLS
+~~~
+
+Users can add TLS to the exporter's HTTP Client by providing certificate
+and key files in the ``tls_config`` parameter.
+
+Supported Aggregators
+---------------------
+
+-  Sum
+-  MinMaxSumCount
+-  Histogram
+-  LastValue
+-  ValueObserver
+
+Error Handling
+--------------
+
+In general, errors are raised by the calling function. The exception is
+for failed requests where any error status code is logged as a warning
+instead.
+
+This is because the exporter does not implement any retry logic as it
+sends cumulative metrics data. This means that data will be preserved
+even if some exports fail.
+
+For example, consider a situation where a user increments a Counter
+instrument 5 times and an export happens between each increment. If the
+exports happen like so:
 
 ::
 
-    pip install opentelemetry-exporter-prometheus-remote-write
+   SUCCESS FAIL FAIL SUCCESS SUCCESS
+   1       2    3    4       5
 
+Then the recieved data will be:
 
-.. _Prometheus: https://prometheus.io/
-.. _OpenTelemetry: https://github.com/open-telemetry/opentelemetry-python/
+::
 
+   1 4 5
 
-References
-----------
+The end result is the same since the aggregations are cumulative
+Contributing
+------------
+
+This exporter's datapath is as follows:
+
+|Exporter datapath| *Entites with ``*`` after their name are not actual
+classes but rather logical groupings of functions within the exporter.*
+
+If you would like to learn more about the exporter's structure and
+design decisions please view the design document below
+
+Design Doc
+~~~~~~~~~~
+
+`Design Document`_
+
+This document is stored elsewhere as it contains large images which will
+significantly increase the size of this repo.
 
-* `Prometheus <https://prometheus.io/>`_
-* `OpenTelemetry Project <https://opentelemetry.io/>`_
+.. _Design Document: https://github.com/open-o11y/docs/tree/master/python-prometheus-remote-write
+.. |Exporter datapath| image:: https://user-images.githubusercontent.com/20804975/100285717-604c7280-2f3f-11eb-9b73-bdf70afce9dd.png
+.. _OTLP: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/protocol/otlp.md
+.. _push controller: https://github.com/open-telemetry/opentelemetry-python/blob/master/opentelemetry-sdk/src/opentelemetry/sdk/metrics/export/controller.py
```

Comment:
Can we link to the actual line number?

Reply:
linked to [L#22](https://github.com/open-telemetry/opentelemetry-python/blob/master/opentelemetry-sdk/src/opentelemetry/sdk/metrics/export/controller.py#L22